src/fides/api/alembic/migrations/versions/fd52d5f08c17_system_dictionary_data_migration.py

```diff
@@ -0,0 +1,374 @@
+
+from fides.config import CONFIG
+
+revision = "fd52d5f08c17"
+down_revision = "39397820a0d5"
+branch_labels = None
+depends_on = None
+
+JOINT_CONTROLLER_INFO = "joint_controller_info"
+DOES_INTERNATIONAL_TRANSFERS = "does_international_transfers"
+RESPONSIBILITY = "responsibility"
+REQUIRES_DATA_PROTECTION_ASSESSMENTS = "requires_data_protection_assessments"
+DPA_LOCATION = "dpa_location"
+DPA_PROGRESS = "dpa_progress"
+
+LEGAL_BASIS_FOR_PROCESSING = "legal_basis_for_processing"
+IMPACT_ASSESSMENT_LOCATION = "impact_assessment_location"
+PROCESSES_SPECIAL_CATEGORY_DATA = "processes_special_category_data"
+SPECIAL_CATEGORY_LEGAL_BASIS = "special_category_legal_basis"
+DATA_SHARED_WITH_THIRD_PARTIES = "data_shared_with_third_parties"
+THIRD_PARTIES = "third_parties"
+RETENTION_PERIOD = "retention_period"
+
+
+def _system_transform_joint_controller_info(
+    joint_controller_decrypted: Optional[Dict],
+    datasets_joint_controller_decrypted: List[Optional[Dict]],
+) -> Optional[str]:
+    """Migrating systems.joint_controller and datasets.joint_controller encrypted json string fields to
+    System.joint_controller_info which is just a string.
+
+    Prioritizing system joint_controller info if it exists, or taking the first non-null entry from datasets
+    otherwise.
+    """
+    new_joint_controller_info: Optional[str] = None
+    # Prioritize system data over dataset data
+    joint_controller_data = (
+        joint_controller_decrypted
+        or next(  # Multiple datasets can be related to a system, so we use the first non null one if applicable
+            (
+                item
+                for item in datasets_joint_controller_decrypted or []
+                if item is not None
+            ),
+            None,
+        )
+    )
+    if joint_controller_data and isinstance(joint_controller_data, dict):
+        separated_fields = [
+            joint_controller_data.get("name"),
+            joint_controller_data.get("address"),
+            joint_controller_data.get("email"),
+            joint_controller_data.get("phone"),
+        ]
+
+        new_joint_controller_info = "; ".join(
+            [elem for elem in separated_fields if elem and elem != "N/A"]
+        )
+        if not new_joint_controller_info:
+            # If this is an empty string, let's set it back to None
+            new_joint_controller_info = None
+
+    return new_joint_controller_info
+
+
+def _system_calculate_does_international_transfers(
+    system_third_country_transfers: Optional[List[str]],
+    dataset_third_country_transfers: Optional[List[str]],
+) -> bool:
+    """Migrating System.third_country_transfers and Dataset.third_country_transfers to System.does_international_transfers
+    If any countries are listed, regardless of which country it is, System.does_international_transfers will be set to true.
+    """
+    return bool(system_third_country_transfers) or bool(dataset_third_country_transfers)
```

Comment:
Nit: `any` seems better here?

```suggestion
    return any((system_third_country_transfers, dataset_third_country_transfers))
```

Reply:
that reads better, thanks!